packages/container/lib/container.js

```diff
@@ -421,15 +421,16 @@ function deprecatedFactoryFor(container, fullName, options = {}) {
   }
 
   let cache = container.factoryCache;
-  if (cache[fullName]) {
-    return cache[fullName];
+  let cached = cache[fullName];
+  if (cached) {
+    return cached;
   }
   let factory = registry.resolve(fullName);
   if (factory === undefined) {
     return;
   }
 
-  let type = fullName.split(':')[0];
+  let [type] = fullName.split(':');
```

Comment:
I think it infers from square brackets that it is array :P, https://cl.ly/0D1F3j103F42